hca-foundation/juniorachievement
# Connecting Backend

# Description
**New**
- POSTs fully functional
- Logic to score and then save assessments to DB
- Created handleConcatenation() func to format user's input for POST request

**Fix**
- Added conditionals to handleConcatenation() to handle when user doesn't correctly fill in a blank so that 'undefined' isn't concatenated and included in request object
- Implemented .lower() on user responses before grading to ensure users aren't penalized for capitalization errors
- Removed low quality images on CheckSlip and DepositForm components and replaced with h2 to match original assessment images
- - Added a 'student_id' field to Assessment model, added that field in serializer.py, and concatenated 'last_name', 'birth_date', 'teacher_name', and the school id in the create function on views.py to create a more unique identifier(note, I kept those individual fields on the model in case they're needed at some point)

**Refactor**
- Consolidated PreAssessment and PostAssessment models into one Assessment model that has a boolean field to determine if it's a pre or post-test
- Initialized objects' keys in initial state
- Updated validatePage() to reference keys in state objects instead of DOM elements

**Style**
- Added borders, background color, and padding to likert components
- Centered CheckSlip, DepositForm, & RegisterSlip components and removed scrollbar visibility(but kept functionality)
- Centered radio buttons on likert components
- Rounded corners of buttons, changed color to the darker green brand color, and adjusted margins to improve spacing

# Testing Instructions
- Try submitting an assessment and check DB to ensure it was added
- Ensure answers are captured and scored accurately
- Check overall styling
